c++/src/capnp/compiler/capnp.c++

```diff
@@ -378,6 +399,53 @@ public:
     return true;
   }
 
+  // Adds a schema file to our `schemaLoader`.
+  // Will resolve all references within the schema.
+  // This is primarily useful for cases such as checking backwards compatibility
+  // where the additional safety checks provided by the compiler are undesirable
+  // (e.g. we want to load two _different_ schema files that happen to have the same
+  // Id in order to compare two versions of the schema.)
+  //
+  // If a schema is loaded that is incompatible with an already loaded schema, this
+  // function will fail, which is itself enough for a surface-level check of backwards
+  // compatibility, where the most recently loaded schema is not compatible with all
+  // previously loaded schemas.
+  //
+  // Note that only types in the top-level schema and types relied on by the top-level
+  // schema will be validated. For example, if we add `a.capnp` which depends on `Blorp`
+  // form `b.capnp` then the compatibility of `Blorp` will be checked when a newer version
+  // of `a.capnp` is loaded (and imports a newer version of `b.capnp`). However, if
+  // `b.capnp` also declares `Blarp` but `a.capnp` does not use `Blarp` then `Blarp`
+  // will not be evaluated for compatibility.
+  kj::MainBuilder::Validity addSchema(kj::StringPtr file) {
+    // schemaLoader is lazy-loaded as it's not needed for all commands.
+    // We _could_ use the compiler here and reach into it for its schema loader
+    // but we are then dependent on the implementation of the compiler when
+    // we don't need to be.
+    if (schemaLoader == kj::none) {
+      schemaLoader.emplace();
+    }
+
+    KJ_IF_SOME(loader, schemaLoader) {
+      // Now we start to actually load the schema files.
+      capnp::SchemaParser parser;
+      DirPathPair dirPathPair = interpretSourceFile(file);
+      parser.parseFromDirectory(dirPathPair.dir, dirPathPair.path.clone(), nullptr);
```

Comment:
Hmm, so does passing `nullptr` as the import path mean this tool doesn't work with schemas that have imports?